sources/web/datalab/polymer/modules/template-manager/template-manager.ts

```diff
@@ -93,6 +100,15 @@ class NotebookTemplate {
   }
 }
 
+// A small hack to allow us to have a unified way of expressing
+// a file location on any of Drive, Jupyter, or our API.
+class ApiFileId extends DatalabFileId {
```

Comment:
I don't understand why we need this. Why not just use a `DatalabFileId`? It'll gladly accept `api` as the source, we just need to add it as a `FileManagerType`.

Reply:
Done.